scvi/inference/posterior.py

```diff
@@ -1258,9 +1258,44 @@ def get_stats(self):
         return libraries.ravel()
 
     @torch.no_grad()
-    def get_sample_scale(self, transform_batch=None):
+    def get_sample_scale(
+        self,
+        transform_batch=None,
+        gene_list=None,
+        library_size=1,
+        batch_size=128,
+        return_df=False,
+        n_samples=1,
+        return_mean=True
+    ):
+        ''' Returns the frequencies of expression for the data.
+
+        :param transform_batch: Batches to condition on.
+        If transform_batch is:
+            - None, then real observed batch is used
+            - int, then batch transform_batch is used
+            - list of int, then px_rates are averaged over provided batches.
+        :param gene_list: (optional) Return frequencies of expression for a subset of genes.
+            This can save memory when working with large datasets and few genes are
+            of interest. (default=None)
+        :param library_size: (optional) Scale the expression frequencies to a common library size.
+            This allows gene expression levels to be interpreted on a common scale of relevant
+            magnitude (default=1, no scaling).
+        :param batch_size: Size of minibatches of the data to use. When only accessing a single
+            gene a larger batch size (e.g. 10000) substantially reduces the time spent on GPU
+            data transfer (default=128).
+        :param return_df: (optional) Return a DataFrame instead of an np.Array. Includes gene
+            names as columns. Requires either n_samples=1 or return_mean=True (default=False).
+        :param n_samples: (optional) Get sample scale from multiple samples (default=1).
+        :param return_mean: (optional) Whether to return the mean of the samples (default=True).
+        '''
+        if gene_list is None:
+            gene_mask = slice(None)
+        else:
+            gene_mask = self.gene_dataset._get_genes_filter_mask_by_attribute(gene_list, return_data=False)
+
         px_scales = []
-        for tensors in self:
+        for tensors in self.sequential(batch_size):
```

Comment:
To be consistent with the other functions, I'd leave this as `for tensors in self`, since the posterior itself can be updated. Also, this function is central to DE, so to be honest I'd have to clear up exactly how this affects everything else.